### PR TITLE
SplitIntervals: add arguments to control output file naming (Issue 7157)

### PR DESCRIPTION
**Pull request by gileshall**
Added two new optional flags to `SplitIntervals`, and their corresponding tests.

1. `--prefix` for adding a prefix to the created interval files
2. `--digits` for configuring the number of digits used to enumerate the interval files.

This modifications were requested by a community user (#7157)

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java [hunk 1]
```diff
@@ -86,11 +86,14 @@
 
     public static final String MIN_CONTIG_SIZE_LONG_NAME = "min-contig-size";
 
+    public static final String INTERVAL_FILE_PREFIX_FULL_NAME = "prefix";
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java:89]: Recommend a slightly more descriptive full name for this argument, such as `interval-file-prefix`

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java [hunk 2]
```diff
@@ -86,11 +86,14 @@
...
     public static final String INTERVAL_FILE_EXTENSION_FULL_NAME = "extension";
 
+    public static final String DEFAULT_PREFIX = "";
     public static final String PICARD_INTERVAL_FILE_EXTENSION = "interval_list";
     public static final String DEFAULT_EXTENSION = "-scattered." + PICARD_INTERVAL_FILE_EXTENSION;
 
+    public static final String INTERVAL_NUMBER_OF_DIGITS_FULL_NAME = "digits";
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java:96]: I'd also suggest a more descriptive full name for this argument, such as `interval-file-num-digits`

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java [hunk 3]
```diff
@@ -108,9 +111,15 @@
...
     public File outputDir;
 
+    @Argument(doc = "Prefix to use when writing interval files", fullName = INTERVAL_FILE_PREFIX_FULL_NAME, optional = true)
+    public String prefix = DEFAULT_PREFIX;
+
     @Argument(doc = "Extension to use when writing interval files", fullName = INTERVAL_FILE_EXTENSION_FULL_NAME, optional = true)
     public String extension = DEFAULT_EXTENSION;
 
+    @Argument(doc = "Number of digits to use when writing interval files", fullName = INTERVAL_NUMBER_OF_DIGITS_FULL_NAME, optional = true)
+    public int numDigits = DEFAULT_NUMBER_OF_DIGITS;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java:121]: Do some up-front validation on this argument value to make sure it's > 0, and throw a `UserException` if it's not.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java [hunk 4]
```diff
@@ -108,9 +111,15 @@
             shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME)
     public File outputDir;
 
+    @Argument(doc = "Prefix to use when writing interval files", fullName = INTERVAL_FILE_PREFIX_FULL_NAME, optional = true)
+    public String prefix = DEFAULT_PREFIX;
+
     @Argument(doc = "Extension to use when writing interval files", fullName = INTERVAL_FILE_EXTENSION_FULL_NAME, optional = true)
     public String extension = DEFAULT_EXTENSION;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java:118]: Does this existing argument satisfy the user's request in https://github.com/broadinstitute/gatk/issues/7157 for an "additional suffix" arg?

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervalsIntegrationTest.java [hunk 5]
```diff
@@ -92,8 +110,27 @@ public void testTwoIntervals() {
...
+                .addInterval("20:1000000-2000000")
+                .addInterval("20:3000000-4000000")
+                .addReference(b37Reference)
+                .add(SplitIntervals.SCATTER_COUNT_SHORT_NAME, scatterCount)
+                .add(SplitIntervals.INTERVAL_NUMBER_OF_DIGITS_FULL_NAME, numDigits)
+                .addOutput(outputDir);
+
+        runCommandLine(args);
+        verifyScatteredFilesExist(scatterCount, outputDir, numDigits, SplitIntervals.DEFAULT_PREFIX, SplitIntervals.DEFAULT_EXTENSION);
+        checkIntervalSizes(scatterCount, outputDir, 2000000, numDigits, SplitIntervals.DEFAULT_PREFIX, SplitIntervals.DEFAULT_EXTENSION);
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervalsIntegrationTest.java:133]: The new integration tests look good

**Review by droazen — CHANGES_REQUESTED** [review]
@gileshall Back to you with a few comments

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java (hunk 1 above)
Comment (gileshall) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java:89]: done

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java (hunk 2 above)
Comment (gileshall) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java:96]: done

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java (hunk 3 above)
Comment (gileshall) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/SplitIntervals.java:121]: good catch.  I populated the `minValue` parameter for the `@Argument` annotation.

**Comment by gileshall** [comment]
Due to my initial forking, I was struggling to get this branch to retest and ended up rebasing unintentionally.  I'm going to close this PR and restart it with an internal branch (ie. not forked).

**Pull request closed without merging**